### DISCUSSION TITLE
Streaming file list support?

### DISCUSSION THREAD

**Discussion by infokiller** [BurntSushi/ripgrep#1639] [General] [answered]
Hi!

Does ripgrep support a streaming list of files to search in?

For example, say I have a (possibly slow) command `list-files` that outputs a list of files to stdout, and I want to search them using rg. I could run `list-files | xargs rg <pattern>`, but then ripgrep will only start running after `list-files` is done (or reaches the maximum args limit). An alternative is to run something like `list-files | xargs -n <max_args> rg <pattern>`, which will invoke ripgrep for every batch of `<max_args>` arguments. The downside of this approach is that a new ripgrep process will be started for each such batch.

**Comment by BurntSushi** [comment, marked as answer]
Nope, ripgrep has no way to do this on its own. You'd have to use `xargs` with `-n`. Starting a new process for each batch doesn't seem to onerous to me, although I grant that it depends on what you're doing.

**Reply by infokiller** [reply]
Yup, it's not too bad, just thought if I can micro-optimize my script a bit further :)
Thanks!
